service/frontend/workflow_handler.go

```diff
@@ -475,6 +475,10 @@ func (wh *WorkflowHandler) prepareStartWorkflowRequest(
 	enums.SetDefaultWorkflowIdReusePolicy(&request.WorkflowIdReusePolicy)
 	enums.SetDefaultWorkflowIdConflictPolicy(&request.WorkflowIdConflictPolicy, enumspb.WORKFLOW_ID_CONFLICT_POLICY_FAIL)
 
+	if err := wh.validateOnConflictOptions(request.OnConflictOptions); err != nil {
+		return nil, err
+	}
```

Comment:
Redundant for now but 🤷

Reply:
Yes, I know is unnecessary for now. Just being preemptive adding the function to validate.